ramanujantools/cmf/cmf.py

```diff
@@ -268,10 +268,32 @@ def walk(
             The limit of the walk multiplication as defined above.
             If `iterations` is a list, returns a list of limits.
         """
-        trajectory_matrix = self.trajectory_matrix(
-            trajectory, start or self.default_origin()
-        )
-        return trajectory_matrix.walk({n: 1}, iterations, {n: 1})
+        iterations_set = set(iterations)
+        if len(iterations_set) != len(iterations):
+            raise ValueError(f"`iterations` values must be unique, got {iterations}")
+
+        if not iterations == sorted(iterations):
```

Comment:
We could also just sort. The existing code is probably better since if the list is not sorted the user is likely making a mistake.

Reply:
Exactly my thought - results coming in a different order than inputs will only results in errors in unaware users. Since they need to sort anyway to match for the result, better have them sort in advance and crash their code in case they are unaware. Always better to have explicit errors than implicit bugs.